virtcontainers/cgroups.go

```diff
@@ -221,31 +221,8 @@ func (s *Sandbox) constrainHypervisor(cgroup cgroups.Cgroup) error {
 		return fmt.Errorf("Could not add hypervisor PID %d to cgroup %v: %v", pid, path, err)
 	}
 
-	// when new container joins, new CPU could be hotplugged, so we
-	// have to query fresh vcpu info from hypervisor for every time.
-	tids, err := s.hypervisor.getThreadIDs()
-	if err != nil {
-		return fmt.Errorf("failed to get thread ids from hypervisor: %v", err)
-	}
-	if tids == nil || len(tids.vcpus) == 0 {
-		// If there's no tid returned from the hypervisor, this is not
-		// a bug. It simply means there is nothing to constrain, hence
-		// let's return without any error from here.
-		return nil
-	}
-
-	// We are about to move just the vcpus (threads) into cgroups with constraints.
-	// Move whole hypervisor process whould be easier but the IO/network performance
-	// whould be impacted.
-	for _, i := range tids.vcpus {
-		// In contrast, AddTask will write thread id to `tasks`
-		// After this, vcpu threads are in "vcpu" sub-cgroup, other threads in
-		// qemu will be left in parent cgroup untouched.
-		if err := cgroup.AddTask(cgroups.Process{
-			Pid: i,
-		}); err != nil {
-			return err
-		}
+	if err := cgroup.Add(cgroups.Process{Pid: pid}); err != nil {
```

Comment:
The pid is also added above to a no constraints cgoup. There is no need to do it anymore.

Reply:
`no constrains cgroup` only adds memory subsystem, to actually limit qemu in the rest of subsystems we need to add to this `cgroup`.

But makes me wonder:  if  it is better to  just add qemu to only to the cpuset subystem and not all the subsystems that has   `cgroup`.? This  way when a container is limited by cpu qouta and but not cpuset will not have performance limitations.

Reply:
Yes, we should not share container CPU quota with the qemu process. Please take a look at https://github.com/kata-containers/runtime/issues/878#issuecomment-477432975.

Reply:
That is an interesting idea. This require  push make k8s a bit more aware of kata containers.  Sounds like a more longer final solution. But always that the ecosystem consider this limitations we have in Kata is better. I can bring this topic to next architecture meeting.

Reply:
Hi @jcvenegas - did this happen? If so, can you summarise the discussion here?

Reply:
Ping @jcvenegas.